File: Deberes/10_Ejercicios/ejercicio_6/Temperatura.cpp
#include <iostream>
#include <limits>
using namespace std;

int main() {
    int temp, suma = 0, contador = 0;
    int frio = 0, templado = 0, calido = 0, muyCalido = 0;
    int mayor = numeric_limits<int>::min();
    int menor = numeric_limits<int>::max();

    cout << "Ingrese temperaturas (999 para terminar):" << endl;
    cin >> temp;

    while (temp != 999) {
        if (temp < -50 || temp > 60) {
            cout << "Dato inválido. Solo entre -50 y 60." << endl;
        } else {
            contador++;
            suma += temp;

            if (temp > mayor) mayor = temp;
            if (temp < menor) menor = temp;

            if (temp < 10) {
                frio++;
            } else if (temp <= 24) {
                templado++;
            } else if (temp <= 34) {
                calido++;
            } else {
                muyCalido++;
            }
        }
        cin >> temp;
    }

    if (contador > 0) {
        double promedio = (double) suma / contador;
        cout << "Cantidad de datos: " << contador << endl;
        cout << "Mayor: " << mayor << endl;
        cout << "Menor: " << menor << endl;
        cout << "Promedio: " << promedio << endl;
        cout << "Fríos: " << frio << endl;
        cout << "Templados: " << templado << endl;
        cout << "Cálidos: " << calido << endl;
        cout << "Muy cálidos: " << muyCalido << endl;
    } else {
        cout << "No se ingresaron temperaturas válidas." << endl;
    }

    return 0;
}

File: Deberes/10_Ejercicios/ejercicio_6/temperatura.java
import java.util.Scanner;

public class RegistroTemperaturas {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int temp, suma = 0, contador = 0;
        int frio = 0, templado = 0, calido = 0, muyCalido = 0;
        int mayor = Integer.MIN_VALUE, menor = Integer.MAX_VALUE;

        System.out.println("Ingrese temperaturas (999 para terminar):");
        temp = sc.nextInt();

        while (temp != 999) {
            if (temp < -50 || temp > 60) {
                System.out.println("Dato inválido. Solo entre -50 y 60.");
            } else {
                contador++;
                suma += temp;

                if (temp > mayor) mayor = temp;
                if (temp < menor) menor = temp;

                if (temp < 10) {
                    frio++;
                } else if (temp <= 24) {
                    templado++;
                } else if (temp <= 34) {
                    calido++;
                } else {
                    muyCalido++;
                }
            }
            temp = sc.nextInt();
        }

        if (contador > 0) {
            double promedio = (double) suma / contador;
            System.out.println("Cantidad de datos: " + contador);
            System.out.println("Mayor: " + mayor);
            System.out.println("Menor: " + menor);
            System.out.println("Promedio: " + promedio);
            System.out.println("Fríos: " + frio);
            System.out.println("Templados: " + templado);
            System.out.println("Cálidos: " + calido);
            System.out.println("Muy cálidos: " + muyCalido);
        } else {
            System.out.println("No se ingresaron temperaturas válidas.");
        }
    }
}
